### ISSUE TITLE
Query parameters are not correctly processed if set in `masterUrl`

### ISSUE THREAD

**Issue by ibuziuk** [fabric8io/kubernetes-client#1089]
```
          String masterUrl = "https://oso-master-url?param=value";

          ConfigBuilder configBuilder =
                  new ConfigBuilder()
                      .withMasterUrl(masterUrl)
                      .withNamespace("namespace)
                      .withTrustCerts(true)
                      .withOauthToken("token");

          KubernetesClient client = new DefaultOpenShiftClient(configBuilder.build());
          List<ServiceAccount> serviceAccounts = client.serviceAccounts().list().getItems();
          for (ServiceAccount sa : serviceAccounts) {
              System.out.println(sa);
          }
```

The sample above fails with the following trace:

> Exception in thread "main" io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: https://oso-master-url/?param=value/api/v1/namespaces/namespace/serviceaccounts. 

Basically, query parameter becomes a url segment and instead of expected `https://oso-master-url/api/v1/namespaces/namespace/serviceaccounts?param=value`, the following url is used while performing requests to oso `https://oso-master-url/?param=value/api/v1/namespaces/namespace/serviceaccounts`. Currently, query parameters are not handled at all while generating `OpenShiftUrl` in KubernetesClient and it simply becomes a url segment.

[1] https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/URLUtils.java#L34

**Comment by hrishin** [comment]
@ibuziuk its limitation to add query string params to the client request. Possible option to send query params is using OKHttpClient request interceptor. (though its kind of overkill to add query parameters)

```java
  @Test
  public void queryParamTest() {
    String masterUrl = "https://MASTER_URL";


    ConfigBuilder configBuilder =
      new ConfigBuilder()
        .withMasterUrl(masterUrl)
        .withNamespace("default")
          .withToken("token");

    Map<String, String> params = new HashMap<String, String>();
    params.put("param1", "value1");

    OkHttpClient httpClient= new OkHttpClient.Builder().addInterceptor(new QueryStringParamInterceptor(params )).build();
    KubernetesClient client = new DefaultKubernetesClient(httpClient, configBuilder.build());

    client.serviceAccounts().list().getItems();


  }

  private class QueryStringParamInterceptor implements Interceptor {

    private final Map<String, String> params;

    public QueryStringParamInterceptor(Map<String, String> params) {
      this.params = params;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
      HttpUrl.Builder builder = chain.request().url()
        .newBuilder();

      for(Map.Entry<String, String> entry : params.entrySet()) {
        builder.addQueryParameter(entry.getKey(), entry.getValue());
      }

      Request request = chain.request().newBuilder().url(builder.build()).build();
      return chain.proceed(request);
    }
  }

```

**Comment by piyush1594** [comment]
Closed in #1104
